src/cs-core/Settings.hpp

```diff
@@ -9,85 +9,166 @@
+  utils::Signal<> const& onSave() const;
+
+  /// Initializes all members from a given JSON file. Once reading finished, the onLoad signal will
+  /// be emitted.
+  void read(std::string const& fileName);
+
+  /// Writes the current settings to a JSON file. Before the state is written to file, the on Save
+  /// signal will be emitted.
+  void write(std::string const& fileName) const;
+
+  // -----------------------------------------------------------------------------------------------
+
+  /// Defines the initial simulation time. Changing this value will have no directly visible effect,
+  /// but any later time resets will use to the new value. Should be either "today" or in the format
+  /// "1950-01-02 00:00:00.000".
+  utils::Property<std::string> pStartDate;
+
+  /// Defines the min and max Date on the timebar. Changing these values will be directly reflected
+  /// in the user interface. Should be in the format "1950-01-02 00:00:00.000".
+  utils::Property<std::string> pMinDate;
+  utils::Property<std::string> pMaxDate;
+
+  /// In order to reduce duplication of code, a list of all used SPICE-frames ("Anchors") is
+  /// required at the start of each configuration file. The name of each Anchor is then later used
+  /// to reference the respective SPICE frame.
   struct Anchor {
     std::string mCenter;
     std::string mFrame;
     std::string mStartExistence;
     std::string mEndExistence;
-  };
 
-  struct Gui {
-    uint32_t mWidthPixel;
-    uint32_t mHeightPixel;
-    double   mWidthMeter;
-    double   mHeightMeter;
-    double   mPosXMeter;
-    double   mPosYMeter;
-    double   mPosZMeter;
-    double   mRotX;
-    double   mRotY;
-    double   mRotZ;
+    /// Convenience method to convert the two strings above to SPICE-compatible doubles.
+    std::pair<double, double> getExistence();
   };
 
+  std::map<std::string, Anchor> mAnchors;
+
+  /// The values of the observer are updated by the SolarSystem once each frame. For all others,
+  /// they should be considered readonly. If you want to modify the transformation of the virtual
+  /// observer, use the api of the SolarSystem instead.
   struct Observer {
-    std::string mCenter;
-    std::string mFrame;
-    double      mLongitude;
-    double      mLatitude;
-    double      mDistance;
-  };
+    /// The name of the SPICE center of the observer.
+    utils::Property<std::string> pCenter;
 
-  struct Location {
-    std::string mPlanet;
-    std::string mPlace;
-  };
+    /// The SPICE frame of reference the observer is currently in.
+    utils::Property<std::string> pFrame;
+
+    /// The position of the observer relative to its center and frame.
+    utils::Property<glm::dvec3> pPosition;
 
+    /// The rotation of the observer relative to its frame.
+    utils::Property<glm::dquat> pRotation;
+  } mObserver;
+
+  /// Events to show on the timenavigation bar
   struct Event {
+    struct Location {
+      std::string mPlanet;
```

Comment:
This probably needs to be changed elsewhere as well.

```suggestion
      std::string mBody;
```

Reply:
I think that I will not touch the Event management in this pull request as we want to refactor this with the minimap/bookmark feature anyways.